src/agent.ts

```diff
@@ -348,4 +342,32 @@ export class Agent {
     this._chat.add(type, message);
     this._logger.message(message);
   }
+
+  private async _executeFunctionIfExists(
+    functionCall: OpenAIFunctionCall
+  ): Promise<void> {
+    const name = functionCall.name!;
+    const functionToCall = (functions(
+      this._library, 
+      this._client
+    ) as any)[name];
+  
+    if (typeof functionToCall !== 'function') {
+      this._logMessage("system", `The model tried to execute a function (${name}) which is not defined in PolyGPT, only "LearnWrap" and "InvokeWrap" can be called.`);
+      // Set the flag to ask the model again
+      this._executedLastIteration = true;
```

Comment:
Why are we setting this "executed last iteration" flag to true if the execution did not happen? (there was an hallucination)